src/Core/ServerSentEvents/ServerEventFilteredChannel.cs

```diff
@@ -31,6 +31,7 @@ namespace SonarLint.VisualStudio.Core.ServerSentEvents
     /// <remarks>
     /// Even though <see cref="IServerSentEventSource{T}"/> and <see cref="IServerSentEventSourcePublisher{T}"/> do not allow their own methods to be called from different threads, this class was designed to allow the calls to two different interfaces' methods to be made from different threads at the same time.
     /// This means that calling from one thread <see cref="GetNextEventOrNullAsync"/> and calling <see cref="Publish"/> or <see cref="Dispose"/> from a different thread at the same time is allowed, while calling methods of the same interface from different threads is not.
+    /// NOTE: there's an exception to this rule that was made to simplify the implementation of the events pump that permits calling Dispose and Publish concurrently, but results in an exception. For more info see <see cref="IServerSentEventSourcePublisher{T}"/>
     /// </remarks>
     /// <typeparam name="T"></typeparam>
     public class ServerEventFilteredChannel<T> : IServerSentEventSource<T>, IServerSentEventSourcePublisher<T> where T : class, IServerEvent
```

Comment:
I'm not sure that the name Filtered is correct here. It doesn't do any filtering. Perhaps just `ServerEventsChannel` ?

Reply:
Well, it's filtered because it has issues that have been filtered, not the ones it needs to filter. But I think removing the word from the name is also fine